qa/tasks/cephfs/test_exports.py

```diff
@@ -198,69 +198,68 @@ def test_ephemeral_pin_dist_override(self):
         """
 
         self._setup_tree(distributed=True)
-        subtrees = self._wait_distributed_subtrees(100, status=self.status, rank="all")
+        subtrees = self._wait_distributed_subtrees(3 * 2, status=self.status, rank="all")
         for s in subtrees:
             path = s['dir']['path']
             if path == '/tree':
-                self.assertEqual(s['export_pin'], 0)
-                self.assertEqual(s['auth_first'], 0)
-            elif path.startswith('/tree/'):
-                self.assertEqual(s['export_pin'], -1)
                 self.assertTrue(s['distributed_ephemeral_pin'])
 
     def test_ephemeral_pin_dist_override_pin(self):
         """
         That an export pin overrides an ephemerally pinned directory.
         """
 
-        self._setup_tree(distributed=True, export=0)
-        subtrees = self._wait_distributed_subtrees(100, status=self.status, rank="all", path="/tree/")
-        which = None
-        for s in subtrees:
-            if s['auth_first'] == 1:
-                path = s['dir']['path']
-                self.mount_a.setfattr(path[1:], "ceph.dir.pin", "0")
-                which = path
-                break
-        self.assertIsNotNone(which)
+        self._setup_tree(distributed=True)
+        subtrees = self._wait_distributed_subtrees(3 * 2, status=self.status, rank="all")
+        self.mount_a.setfattr("tree", "ceph.dir.pin", "0")
```

Comment:
You've changed what the test was checking. We want to check that the immediate child directory (like `/tree/3`) which is ephemerally pinned to rank 1 can be moved to rank 0 via an export pin.

Reply:
I realize the directory is fragmented now... can we load the MDS cache to find a sub-directory is pinned to rank 1 and then export pin that?

Reply:
test_ephemeral_pin_dist_override_before does that

Reply:
Okay, so what is the purpose of this test? Now the test is pinning `/tree` instead of `/tree/<N>`. We expect the fragments to stay pinned now?

Then the test below should just be:

`self._wait_subtrees([('/tree', 0)], status=self.status)`

Reply:
self._wait_subtrees([('/tree', 0)], ...) does not work. we don't know how many frags are there

Reply:
Maybe I'm not understanding the new behavior but we expect 3*2 fragments to be created and distributed after setting the distributed epin on `/tree`. Then you're pinning `/tree` to `0`. That should cause all of the fragments to lose their ephemeral pins and move back to rank 0, right? Then I would expect the subtrees to merge and `/tree` should be the only subtree, pinned to rank 0?

Reply:
merging dirfrags needs a trigger. this is no one